matlab/src/feather_reader.cc

```diff
@@ -242,16 +245,34 @@ mxArray* FeatherReader::ReadVariables() const {
   mxArray* variables =
       mxCreateStructMatrix(1, num_variables_, num_variable_fields, fieldnames);
 
-  // Read all the table variables in the Feather file into memory.
+  // Read the entire table in the Feather file into memory.
+  std::shared_ptr<arrow::Table> table = nullptr;
+  arrow::Status status = reader_->Read(&table);
+  if (!status.ok()) {
+    mexErrMsgIdAndTxt("MATLAB:arrow:FeatherReader::FailedToReadTable",
+                      "Failed to read arrow::Table from Feather file.");
```

Comment:
do you want to append the Status message? You might consider having a utility to adorn a status with additional Matlab-specific error message to make this pattern simpler

Reply:
That's a good point. I'll update the error to include the status message.